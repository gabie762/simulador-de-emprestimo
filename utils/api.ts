// utils/api.ts
import { Produto, Resultado } from '../types';

// Configuração da API
const API_BASE_URL = 'https://sua-api.com/api'; // Substitua pela URL da sua API

// Tipos para requests/responses
export interface CadastrarProdutoRequest {
  nome: string;
  taxaJurosAnual: number;
  prazoMaximo: number;
}

export interface SimulacaoRequest {
  produtoId: string;
  valor: number;
  prazo: number;
}

export interface SimulacaoResponse {
  produto: Produto;
  valorEmprestimo: number;
  prazo: number;
  taxaJurosAnual: number;
  taxaJurosMensal: number;
  valorParcela: number;
  totalPagar: number;
  totalJuros: number;
  memoriaCalculo?: {
    mes: number;
    parcela: number;
    juros: number;
    amortizacao: number;
    saldoDevedor: number;
  }[];
}

// Função helper para fazer requests
async function apiRequest<T>(
  endpoint: string,
  options: RequestInit = {}
): Promise<T> {
  const url = `${API_BASE_URL}${endpoint}`;
  
  const defaultHeaders = {
    'Content-Type': 'application/json',
    'Accept': 'application/json',
    // Adicione headers de autenticação se necessário
    // 'Authorization': `Bearer ${token}`,
  };

  const config: RequestInit = {
    ...options,
    headers: {
      ...defaultHeaders,
      ...options.headers,
    },
  };

  try {
    const response = await fetch(url, config);
    
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}: ${response.statusText}`);
    }
    
    const data = await response.json();
    return data;
  } catch (error) {
    console.error(`Erro na API (${endpoint}):`, error);
    throw error;
  }
}

// GET /produtos – Listar produtos
export async function getProdutos(): Promise<Produto[]> {
  try {
    const produtos = await apiRequest<Produto[]>('/produtos', {
      method: 'GET',
    });
    return produtos;
  } catch (error) {
    console.error('Erro ao buscar produtos:', error);
    throw new Error('Não foi possível carregar os produtos do servidor.');
  }
}

// POST /produtos – Cadastrar produto
export async function postProduto(produto: CadastrarProdutoRequest): Promise<Produto> {
  try {
    const novoProduto = await apiRequest<Produto>('/produtos', {
      method: 'POST',
      body: JSON.stringify(produto),
    });
    return novoProduto;
  } catch (error) {
    console.error('Erro ao cadastrar produto:', error);
    
    // Tratamento de erros específicos
    if (error instanceof Error) {
      if (error.message.includes('409')) {
        throw new Error('PRODUTO_DUPLICADO');
      }
      if (error.message.includes('400')) {
        throw new Error('Dados do produto inválidos.');
      }
    }
    
    throw new Error('Não foi possível cadastrar o produto.');
  }
}

// POST /simulações – Realizar simulação de empréstimo
export async function postSimulacao(simulacao: SimulacaoRequest): Promise<Resultado> {
  try {
    const resultado = await apiRequest<SimulacaoResponse>('/simulacoes', {
      method: 'POST',
      body: JSON.stringify(simulacao),
    });
    
    // Converte o response da API para o formato esperado pelo app
    const resultadoFormatado: Resultado = {
      produto: resultado.produto.nome,
      valorEmprestimo: resultado.valorEmprestimo.toFixed(2),
      prazo: resultado.prazo,
      taxaJurosAnual: resultado.taxaJurosAnual.toFixed(2) + '%',
      taxaJurosMensal: resultado.taxaJurosMensal.toFixed(2) + '%',
      valorParcela: resultado.valorParcela.toFixed(2),
      totalPagar: resultado.totalPagar.toFixed(2),
      totalJuros: resultado.totalJuros.toFixed(2),
      memoriaCalculo: resultado.memoriaCalculo?.map(item => ({
        mes: item.mes,
        parcela: item.parcela.toFixed(2),
        juros: item.juros.toFixed(2),
        amortizacao: item.amortizacao.toFixed(2),
        saldoDevedor: item.saldoDevedor.toFixed(2),
      })) || [],
    };
    
    return resultadoFormatado;
  } catch (error) {
    console.error('Erro ao realizar simulação:', error);
    
    // Tratamento de erros específicos
    if (error instanceof Error) {
      if (error.message.includes('400')) {
        throw new Error('Dados da simulação inválidos.');
      }
      if (error.message.includes('404')) {
        throw new Error('Produto não encontrado.');
      }
    }
    
    throw new Error('Não foi possível realizar a simulação.');
  }
}

// Função para deletar produto (se a API suportar)
export async function deleteProduto(id: string): Promise<boolean> {
  try {
    await apiRequest(`/produtos/${id}`, {
      method: 'DELETE',
    });
    return true;
  } catch (error) {
    console.error('Erro ao deletar produto:', error);
    throw new Error('Não foi possível deletar o produto.');
  }
}

// Função para testar conectividade com a API
export async function testApiConnection(): Promise<boolean> {
  try {
    // Tenta fazer uma requisição simples para verificar se a API está acessível
    await apiRequest('/produtos', { method: 'GET' });
    return true;
  } catch (error) {
    console.error('API não está acessível:', error);
    return false;
  }
}
