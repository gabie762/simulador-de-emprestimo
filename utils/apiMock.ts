// utils/apiMock.ts
import AsyncStorage from '@react-native-async-storage/async-storage';

const STORAGE_KEY = 'produtos_mock';
const defaultProdutos = [
  { id: '1', nome: 'Crédito Pessoal', taxaJurosAnual: 10.5, prazoMaximo: 24 },
  { id: '2', nome: 'Crediário CAIXA', taxaJurosAnual: 8.2, prazoMaximo: 36 },
  { id: '3', nome: 'Empréstimo Consignado', taxaJurosAnual: 6.8, prazoMaximo: 84 },
];

export async function getProdutos() {
  try {
    const stored = await AsyncStorage.getItem(STORAGE_KEY);
    if (stored) {
      return JSON.parse(stored);
    } else {
      await AsyncStorage.setItem(STORAGE_KEY, JSON.stringify(defaultProdutos));
      return [...defaultProdutos];
    }
  } catch (error) {
    console.error('Erro ao carregar produtos:', error);
    return [...defaultProdutos];
  }
}

export async function postProduto(produto: any) {
  return new Promise((resolve, reject) => {
    setTimeout(async () => {
      try {
        const stored = await AsyncStorage.getItem(STORAGE_KEY);
        const produtos = stored ? JSON.parse(stored) : [...defaultProdutos];
        
        // Verifica se já existe um produto com o mesmo nome
        const produtoExistente = produtos.find((p: any) => 
          p.nome.toLowerCase().trim() === produto.nome.toLowerCase().trim()
        );
        
        if (produtoExistente) {
          reject(new Error('PRODUTO_DUPLICADO'));
          return;
        }
        
        const novo = { ...produto, id: Date.now().toString() };
        produtos.push(novo);
        await AsyncStorage.setItem(STORAGE_KEY, JSON.stringify(produtos));
        resolve(novo);
      } catch (error) {
        console.error('Erro ao salvar produto:', error);
        reject(error instanceof Error ? error : new Error('Erro desconhecido'));
      }
    }, 500);
  });
}

export async function resetProdutos() {
  await AsyncStorage.removeItem(STORAGE_KEY);
  await AsyncStorage.setItem(STORAGE_KEY, JSON.stringify(defaultProdutos));
}

export async function deleteProduto(id: string) {
  try {
    const stored = await AsyncStorage.getItem(STORAGE_KEY);
    const produtos = stored ? JSON.parse(stored) : [];
    
    const produtosFiltrados = produtos.filter((produto: any) => produto.id !== id);
    await AsyncStorage.setItem(STORAGE_KEY, JSON.stringify(produtosFiltrados));
    
    return true;
  } catch (error) {
    console.error('Erro ao deletar produto:', error);
    return false;
  }
}
