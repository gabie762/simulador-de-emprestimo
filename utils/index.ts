import { Produto, Resultado } from '../types';
import { Platform } from 'react-native';

export const mockProdutos: Produto[] = [
  { id: '1', nome: 'Crédito Pessoal Padrão', taxaJurosAnual: 25.0, prazoMaximo: 48 },
  { id: '2', nome: 'Crédito Imobiliário', taxaJurosAnual: 9.5, prazoMaximo: 360 },
  { id: '3', nome: 'Financiamento Veicular', taxaJurosAnual: 18.2, prazoMaximo: 60 },
];

// Helper para normalizar estilos shadow para web
export const normalizeShadow = (shadowStyle: any) => {
  if (Platform.OS === 'web') {
    const { shadowColor, shadowOffset, shadowOpacity, shadowRadius, ...rest } = shadowStyle;
    
    if (shadowColor && shadowOffset && shadowOpacity !== undefined && shadowRadius) {
      return {
        ...rest,
        boxShadow: `${shadowOffset.width}px ${shadowOffset.height}px ${shadowRadius}px rgba(0, 0, 0, ${shadowOpacity})`,
      };
    }
  }
  
  return shadowStyle;
};

export const calcularSimulacao = (
  valorEmprestimo: number,
  prazo: number,
  produto: Produto
): Resultado => {
  const taxaMensal = (produto.taxaJurosAnual / 12) / 100;
  const parcela = valorEmprestimo * (taxaMensal * Math.pow(1 + taxaMensal, prazo)) / (Math.pow(1 + taxaMensal, prazo) - 1);
  const total = parcela * prazo;
  const totalJuros = total - valorEmprestimo;
  
  let saldoDevedor = valorEmprestimo;
  const memoriaCalculo = [];
  
  for (let i = 1; i <= prazo; i++) {
    const jurosMes = saldoDevedor * taxaMensal;
    const amortizacao = parcela - jurosMes;
    saldoDevedor -= amortizacao;
    memoriaCalculo.push({ 
      mes: i, 
      parcela: parcela.toFixed(2), 
      juros: jurosMes.toFixed(2), 
      amortizacao: amortizacao.toFixed(2), 
      saldoDevedor: Math.abs(saldoDevedor).toFixed(2) 
    });
  }

  return {
    produto: produto.nome,
    valorEmprestimo: valorEmprestimo.toFixed(2),
    prazo: prazo,
    taxaJurosAnual: produto.taxaJurosAnual.toFixed(2),
    taxaJurosMensal: (taxaMensal * 100).toFixed(4),
    valorParcela: parcela.toFixed(2),
    totalPagar: total.toFixed(2),
    totalJuros: totalJuros.toFixed(2),
    memoriaCalculo,
  };
};
