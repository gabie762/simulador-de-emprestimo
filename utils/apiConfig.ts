// utils/apiConfig.ts
import * as ApiReal from './api';
import * as ApiMock from './apiMock';

// Configuração para alternar entre Mock e API real
export const USE_REAL_API = false; // Mude para true quando quiser usar a API real

// Exporta as funções baseado na configuração
export const getProdutos = USE_REAL_API ? ApiReal.getProdutos : ApiMock.getProdutos;
export const postProduto = USE_REAL_API ? ApiReal.postProduto : ApiMock.postProduto;
export const deleteProduto = USE_REAL_API ? ApiReal.deleteProduto : ApiMock.deleteProduto;

// Função de simulação (só existe na API real, no mock usamos o cálculo local)
export const postSimulacao = USE_REAL_API ? ApiReal.postSimulacao : null;

// Função para testar conectividade (só para API real)
export const testApiConnection = USE_REAL_API ? ApiReal.testApiConnection : null;
