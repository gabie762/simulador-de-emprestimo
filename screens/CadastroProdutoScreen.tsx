import React, { useEffect, useState } from 'react';
import { ActivityIndicator, Animated, ScrollView, StyleSheet, Text, TextInput, TouchableOpacity, View } from 'react-native';
import { globalStyles } from '../styles';
import { postProduto } from '../utils/apiConfig';

interface CadastroProdutoScreenProps {
  goHome: () => void;
  atualizarProdutos: () => Promise<void>;
  testID?: string;
}

const CadastroProdutoScreen: React.FC<CadastroProdutoScreenProps> = ({ goHome, atualizarProdutos, testID }) => {
  const [nome, setNome] = useState('');
  const [taxaJurosAnual, setTaxaJurosAnual] = useState('');
  const [prazoMaximo, setPrazoMaximo] = useState('');
  const [isSubmitting, setIsSubmitting] = useState(false);
  const [mensagem, setMensagem] = useState<{ texto: string; tipo: 'sucesso' | 'erro' } | null>(null);
  const [fadeAnim] = useState(new Animated.Value(0));
  const [slideAnim] = useState(new Animated.Value(-100));

  const limparCampos = () => {
    setNome('');
    setTaxaJurosAnual('');
    setPrazoMaximo('');
    setMensagem(null);
  };

  const mostrarToast = (texto: string, tipo: 'sucesso' | 'erro') => {
    setMensagem({ texto, tipo });
    
    // Reset das animações
    fadeAnim.setValue(0);
    slideAnim.setValue(-50);
    
    // Animação de entrada
    Animated.parallel([
      Animated.timing(fadeAnim, {
        toValue: 1,
        duration: 300,
        useNativeDriver: true,
      }),
      Animated.timing(slideAnim, {
        toValue: 0,
        duration: 300,
        useNativeDriver: true,
      }),
    ]).start();

    // Auto-hide após 3 segundos
    setTimeout(() => {
      esconderToast();
    }, 3000);
  };

  const esconderToast = () => {
    Animated.parallel([
      Animated.timing(fadeAnim, {
        toValue: 0,
        duration: 300,
        useNativeDriver: true,
      }),
      Animated.timing(slideAnim, {
        toValue: -100,
        duration: 300,
        useNativeDriver: true,
      }),
    ]).start(() => {
      setMensagem(null);
    });
  };

  useEffect(() => {
    limparCampos();
  }, []);

  const handleVoltar = () => {
    limparCampos();
    goHome();
  };

  const handleCadastro = async () => {
    if (!nome || !taxaJurosAnual || !prazoMaximo) {
      mostrarToast('Todos os campos são obrigatórios.', 'erro');
      return;
    }
    setIsSubmitting(true);
    setMensagem(null);
    const novoProduto = { 
      nome, 
      taxaJurosAnual: parseFloat(taxaJurosAnual.replace(',', '.')), 
      prazoMaximo: parseInt(prazoMaximo) 
    };
    
    try {
      await postProduto(novoProduto);
      
      await atualizarProdutos();
      
      // Limpa os campos após atualizar os produtos
      limparCampos();
      
      // Mostra toast de sucesso
      mostrarToast('Produto cadastrado com sucesso!', 'sucesso');
      
    } catch (error) {
      console.error("Erro ao cadastrar produto:", error);
      if (error instanceof Error && error.message === 'PRODUTO_DUPLICADO') {
        mostrarToast('Já existe um produto cadastrado com este nome. Escolha um nome diferente.', 'erro');
      } else {
        mostrarToast('Não foi possível conectar ao servidor para cadastrar o produto.', 'erro');
      }
    } finally {
      setIsSubmitting(false);
    }
  };

  return (
    <View testID={testID}>
      <ScrollView 
        style={{ flex: 1 }} 
        contentContainerStyle={[globalStyles.container, { 
          flexGrow: 1,
          paddingBottom: 120 // Espaço extra para o tab bar
        }]}
        showsVerticalScrollIndicator={false}
        keyboardShouldPersistTaps="handled"
        bounces={false}
      >
        <Text style={globalStyles.title}>Cadastrar Novo Produto</Text>
        
        {/* Toast Notification */}
        {mensagem && (
          <Animated.View 
            style={[
              styles.toast,
              mensagem.tipo === 'sucesso' ? styles.toastSucesso : styles.toastErro,
              {
                opacity: fadeAnim,
                transform: [{ translateY: slideAnim }]
              }
            ]}
          >
            <Text style={styles.toastTexto}>{mensagem.texto}</Text>
          </Animated.View>
        )}
        
        <TextInput 
          style={globalStyles.input} 
          placeholder="Nome do Produto" 
          placeholderTextColor="#888" 
          value={nome} 
          onChangeText={setNome}
        />
        <TextInput 
          style={globalStyles.input} 
          placeholder="Taxa de Juros Anual (%)" 
          placeholderTextColor="#888" 
          keyboardType="numeric" 
          value={taxaJurosAnual} 
          onChangeText={setTaxaJurosAnual}
        />
        <TextInput 
          style={globalStyles.input} 
          placeholder="Prazo Máximo (meses)" 
          placeholderTextColor="#888" 
          keyboardType="numeric" 
          value={prazoMaximo} 
          onChangeText={setPrazoMaximo}
        />
        <TouchableOpacity 
          style={[globalStyles.buttonOrange, isSubmitting && globalStyles.buttonDisabled]} 
          onPress={handleCadastro} 
          disabled={isSubmitting}
        >
          {isSubmitting ? 
            <ActivityIndicator color="#fff" /> : 
            <Text style={globalStyles.buttonText}>Cadastrar</Text>
          }
        </TouchableOpacity>
        <TouchableOpacity style={[globalStyles.button, globalStyles.buttonSecondary]} onPress={handleVoltar}>
          <Text style={globalStyles.buttonTextSecondary}>Voltar</Text>
        </TouchableOpacity>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  toast: {
    position: 'absolute',
    top: 50,
    left: 20,
    right: 20,
    padding: 15,
    borderRadius: 10,
    zIndex: 1000,
    elevation: 10,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.25,
    shadowRadius: 4,
  },
  toastSucesso: {
    backgroundColor: '#4CAF50',
  },
  toastErro: {
    backgroundColor: '#F44336',
  },
  toastTexto: {
    color: '#fff',
    fontSize: 16,
    textAlign: 'center',
    fontFamily: 'Caixa',
    fontWeight: 'bold',
  },
});

export default CadastroProdutoScreen;