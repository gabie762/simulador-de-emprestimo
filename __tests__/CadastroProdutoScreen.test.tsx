import { fireEvent, render, waitFor } from '@testing-library/react-native';
import React from 'react';
import CadastroProdutoScreen from '../screens/CadastroProdutoScreen';
import { postProduto } from '../utils/apiConfig';

jest.mock('../utils/apiConfig', () => ({
  postProduto: jest.fn(),
}));

describe('CadastroProdutoScreen', () => {
  beforeEach(() => {
    jest.useFakeTimers();
    jest.spyOn(global.console, 'error').mockImplementation(() => {});
  });

  afterEach(() => {
    jest.runAllTimers();
    jest.restoreAllMocks();
  });

  it('renders correctly', () => {
    const mockGoHome = jest.fn();
    const mockAtualizarProdutos = jest.fn();
    const { toJSON } = render(
      <CadastroProdutoScreen goHome={mockGoHome} atualizarProdutos={mockAtualizarProdutos} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('shows error message when fields are empty', async () => {
    const mockGoHome = jest.fn();
    const mockAtualizarProdutos = jest.fn();
    const { getByText } = render(
      <CadastroProdutoScreen goHome={mockGoHome} atualizarProdutos={mockAtualizarProdutos} />
    );

    fireEvent.press(getByText('Cadastrar'));

    await waitFor(() => {
      expect(getByText('Todos os campos são obrigatórios.')).toBeTruthy();
    });
  });

  it('submits successfully and shows success message', async () => {
    const mockGoHome = jest.fn();
    const mockAtualizarProdutos = jest.fn();
    (postProduto as jest.Mock).mockResolvedValueOnce(undefined);

    const { getByText, getByPlaceholderText } = render(
      <CadastroProdutoScreen goHome={mockGoHome} atualizarProdutos={mockAtualizarProdutos} />
    );

    fireEvent.changeText(getByPlaceholderText('Nome do Produto'), 'Produto Teste');
    fireEvent.changeText(getByPlaceholderText('Taxa de Juros Anual (%)'), '5');
    fireEvent.changeText(getByPlaceholderText('Prazo Máximo (meses)'), '12');
    fireEvent.press(getByText('Cadastrar'));

    await waitFor(() => {
      expect(getByText('Produto cadastrado com sucesso!')).toBeTruthy();
      expect(mockAtualizarProdutos).toHaveBeenCalled();
    });
  });

  it('shows error message on submission failure', async () => {
    const mockGoHome = jest.fn();
    const mockAtualizarProdutos = jest.fn();
    (postProduto as jest.Mock).mockRejectedValueOnce(new Error('PRODUTO_DUPLICADO'));

    const { getByText, getByPlaceholderText } = render(
      <CadastroProdutoScreen goHome={mockGoHome} atualizarProdutos={mockAtualizarProdutos} />
    );

    fireEvent.changeText(getByPlaceholderText('Nome do Produto'), 'Produto Teste');
    fireEvent.changeText(getByPlaceholderText('Taxa de Juros Anual (%)'), '5');
    fireEvent.changeText(getByPlaceholderText('Prazo Máximo (meses)'), '12');
    fireEvent.press(getByText('Cadastrar'));

    await waitFor(() => {
      expect(
        getByText('Já existe um produto cadastrado com este nome. Escolha um nome diferente.')
      ).toBeTruthy();
    });
  });

  it('calls goHome when Voltar is pressed', () => {
    const mockGoHome = jest.fn();
    const mockAtualizarProdutos = jest.fn();

    const { getByText } = render(
      <CadastroProdutoScreen goHome={mockGoHome} atualizarProdutos={mockAtualizarProdutos} />
    );

    fireEvent.press(getByText('Voltar'));

    expect(mockGoHome).toHaveBeenCalled();
  });
});
