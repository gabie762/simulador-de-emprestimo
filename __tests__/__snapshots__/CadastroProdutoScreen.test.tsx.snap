// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CadastroProdutoScreen renders correctly 1`] = `
<View>
  <RCTScrollView
    bounces={false}
    contentContainerStyle={
      [
        {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "justifyContent": "flex-start",
          "maxWidth": 600,
          "padding": 20,
          "width": "100%",
        },
        {
          "flexGrow": 1,
          "paddingBottom": 120,
        },
      ]
    }
    keyboardShouldPersistTaps="handled"
    showsVerticalScrollIndicator={false}
    style={
      {
        "flex": 1,
      }
    }
  >
    <View>
      <Text
        style={
          {
            "color": "#0066B3",
            "fontFamily": "Caixa",
            "fontSize": 28,
            "fontWeight": "bold",
            "marginBottom": 20,
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        Cadastrar Novo Produto
      </Text>
      <TextInput
        onChangeText={[Function]}
        placeholder="Nome do Produto"
        placeholderTextColor="#888"
        style={
          {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#C5C7C8",
            "borderRadius": 8,
            "borderWidth": 1,
            "color": "#0066B3",
            "fontFamily": "Caixa",
            "fontSize": 16,
            "marginBottom": 15,
            "paddingHorizontal": 15,
            "paddingVertical": 12,
            "width": "100%",
          }
        }
        value=""
      />
      <TextInput
        keyboardType="numeric"
        onChangeText={[Function]}
        placeholder="Taxa de Juros Anual (%)"
        placeholderTextColor="#888"
        style={
          {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#C5C7C8",
            "borderRadius": 8,
            "borderWidth": 1,
            "color": "#0066B3",
            "fontFamily": "Caixa",
            "fontSize": 16,
            "marginBottom": 15,
            "paddingHorizontal": 15,
            "paddingVertical": 12,
            "width": "100%",
          }
        }
        value=""
      />
      <TextInput
        keyboardType="numeric"
        onChangeText={[Function]}
        placeholder="Prazo MÃ¡ximo (meses)"
        placeholderTextColor="#888"
        style={
          {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#C5C7C8",
            "borderRadius": 8,
            "borderWidth": 1,
            "color": "#0066B3",
            "fontFamily": "Caixa",
            "fontSize": 16,
            "marginBottom": 15,
            "paddingHorizontal": 15,
            "paddingVertical": 12,
            "width": "100%",
          }
        }
        value=""
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#F7941E",
            "borderRadius": 8,
            "marginBottom": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "Caixa",
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          Cadastrar
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#0066B3",
            "borderRadius": 8,
            "borderWidth": 1,
            "marginBottom": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#0066B3",
              "fontFamily": "Caixa",
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          Voltar
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
